pkg/checks/traceroute/traceroute.go

```diff
@@ -126,44 +176,69 @@ func TraceRoute(ctx context.Context, cfg tracerouteConfig) (map[int][]Hop, error
 	var wg sync.WaitGroup
 
 	for ttl := 1; ttl <= cfg.MaxHops; ttl++ {
+		c, hopSpan := tr.Start(ctx, fmt.Sprintf("Hop %d", ttl), trace.WithAttributes(
+			attribute.Int("ttl", ttl),
+		))
 		wg.Add(1)
 		go func(ttl int) {
 			defer wg.Done()
+			defer hopSpan.End()
+
 			l := log.With("ttl", ttl)
-			logctx := logger.IntoContext(ctx, l)
+			logctx := logger.IntoContext(c, l)
+
+			retry := 0
 			err := helper.Retry(func(ctx context.Context) error {
+				defer func() {
+					retry++
+				}()
+				hopSpan.AddEvent("Attempting to hop", trace.WithAttributes(
+					attribute.Int("ttl", ttl),
+					attribute.Int("retry", retry),
+				))
+
 				hop, err := traceroute(ctx, addr, ttl, cfg.Timeout)
 				if hop != nil {
 					results <- *hop
 				}
 				if err != nil {
-					l.Error("traceroute failed", "err", err.Error())
+					l.ErrorContext(ctx, "Failed to trace route", "err", err.Error())
+					hopSpan.RecordError(err)
 					return err
 				}
 				if !hop.Reached {
-					l.Debug("failed to reach target, retrying")
+					l.DebugContext(ctx, "failed to reach target, retrying")
 					return errors.New("failed to reach target")
 				}
 				return nil
 			}, cfg.Rc)(logctx)
 			if err != nil {
```

Comment:
Is there no need to record the error here as well? I would argue that if the target couldn't be reached, the span should display this information as well with `RecordError`